packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -496,13 +496,17 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
   }
 
   @autobind
-  private _onMenuKeyDown(ev: React.KeyboardEvent<HTMLElement>) {
+  private _onMenuKeyDown(ev: React.KeyboardEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) {
     if (ev.which === KeyCodes.down) {
       let { onMenuClick } = this.props;
       onMenuClick && onMenuClick(ev, this);
       !ev.defaultPrevented && this._onToggleMenu();
       ev.preventDefault();
       ev.stopPropagation();
+    } else {
+      if (this.props.onKeyDown) {
```

Comment:
Can you just move 507 to line 500, outside of the if? I think onKeyDown should always be executed if passed in.

Additionally, I believe if `ev.defaultPrevented` is true, we should maybe skip our default logic.

Reply:
Yes can do :)